yt/data_objects/construction_data_containers.py

```diff
@@ -899,6 +913,36 @@ def _fill_fields(self, fields):
             fi = self.ds._get_field_info(field)
             self[field] = self.ds.arr(dest, fi.units)
 
+    def _fill_sph_particles(self, fields):
+        # checks that we have the field and gets information
+        fields = [f for f in fields if f not in self.field_data]
+        if len(fields) == 0: return
+
+        ptype = self.ds._sph_ptype
+        for field in fields:
+            dest = np.zeros(self.ActiveDimensions, dtype="float64")
+
+            for chunk in self._data_source.chunks([field], "io"):
```

Comment:
what if we put the progress bar in this loop? I don't know if we necessarily know the number of chunks ahead of time, but at least we can update the progress of a single progress bar instead of creating a bunch of them in the cython code this for loop calls.

Reply:
do you mean start the progess bar outside the loop and update from within cython? otherwise we will only update every chunk which may be very slow and very few updates

Reply:
Yes, you can pass the progress bar into the cython routine. If `get_pbar` isn't sufficient for your needs, you can just use our vendored copy of `tqdm` directly (e.g. [here](https://github.com/tqdm/tqdm#manual), perhaps without the `total` keyword if we don't know the number of particles we're smoothing ahead of time). Our vendored copy of `tqdm` lives in `yt.extern.tqdm`.